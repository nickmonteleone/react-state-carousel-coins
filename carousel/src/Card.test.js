import React from "react";
import { render } from "@testing-library/react";
import Card from "./Card";

const testCard = {
  caption: "test card",
  src: "/test/test.png"
};
const testCardIdx = 1;
const totalNum = 3;

it("renders without crashing (smoke test)", function () {
  render(
    <Card
      caption={testCard.caption}
      src={testCard.src}
      currNum={testCardIdx}
      totalNum={totalNum}
    />
  );
});

it("matches snapshot", function () {
  const { container } = render(
    <Card
      caption={testCard.caption}
      src={testCard.src}
      currNum={testCardIdx}
      totalNum={totalNum}
    />
  );
  expect(container).toMatchSnapshot();
});
