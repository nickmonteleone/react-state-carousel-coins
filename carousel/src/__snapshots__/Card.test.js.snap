// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      test card
    </h4>
    <img
      alt="test card"
      class="Card-image"
      src="/test/test.png"
    />
    <small
      class="Card-small"
    >
      Image 
      1
       of 
      3
      .
    </small>
  </div>
</div>
`;
